library/core/src/sync/atomic.rs

```diff
@@ -3326,23 +3326,23 @@ macro_rules! atomic_int {
                 unsafe { atomic_xor(self.v.get(), val, order) }
             }
 
-            /// Fetches the value, and applies a function to it that returns an optional
-            /// new value. Returns a `Result` of `Ok(previous_value)` if the function returned `Some(_)`, else
-            /// `Err(previous_value)`.
+            /// Loads the current value, applies a closure to it, and optionally tries to store a new value.
             ///
-            /// Note: This may call the function multiple times if the value has been changed from other threads in
-            /// the meantime, as long as the function returns `Some(_)`, but the function will have been applied
-            /// only once to the stored value.
+            /// If the closure ever returns None, this method will immediately return `Err(current value)`.
```

Comment:
```suggestion
            /// If the closure ever returns `None`, this method will immediately return `Err(loaded_value)`.
```

Reply:
Right, the issue you're identifying is that you can never really be sure whether the value you have is current or not, only that you loaded it.

Current usage is "Loads the current value ..." and "... return `Err(current value)`" plus 3 more. Your suggestion taken literally is "Loads the current value ..." and "... return `Err(loaded_value)`" in which loaded_value seems to not have an antecedent any more. And if I take it more consistently, it only seems to become worse: "Loads the loaded_value ..." and "... return `Err(loaded_value)`".

Your critique is valid, but I don't think this solution works.